gpkit/constraints/relax.py

```diff
@@ -64,14 +76,31 @@ def __init__(self, constraints):
         if not isinstance(constraints, ConstraintSet):
             constraints = ConstraintSet(constraints)
         substitutions = dict(constraints.substitutions)
-        posynomials = constraints.as_posyslt1()
-        N = len(posynomials)
+        relconstrs = []
+        N = len(constraints)
         with NamedVariables("Relax"):
             self.relaxvars = VectorVariable(N, "C")
-        ConstraintSet.__init__(self,
-                               [[posynomials <= self.relaxvars],
-                                self.relaxvars >= 1],
-                               substitutions)
+        i = Count()
+        with SignomialsEnabled():
+            for constr in constraints.flat(constraintsets=False):
+                if constr.oper == ">=":
+                    relconstrs.append(self.relaxvars[i.next()]*constr.left >=
+                                      constr.right)
+                elif constr.oper == "<=":
+                    relconstrs.append(constr.left <=
+                                      self.relaxvars[i.next()]*constr.right)
+                elif constr.oper == "=":
+                    relconstrs.append(constr.left <=
+                                      self.relaxvars[i.next()]*constr.right)
+                    relconstrs.append(self.relaxvars[i.count]*constr.left >=
+                                      constr.right)
+                else:
```

Comment:
there's a lot of duplication of code between this and the above; consider making a new function that handles the constraint addition?

and instead of using Counter, I'd do `for i, constr in enumerate(constraints.flat(constraintsets=False)):`